### PR TITLE
Support Rename using the Controller

### PR DESCRIPTION
**Pull request by TMaster**
Hi @rychkog

Regarding issue [129](https://github.com/valor-software/ng2-tree/issues/129).

I wrote a cool feature allowing the user to rename certain node via the TreeController api.

I've been using this function for quit some time and it works fine.

Care to merge?

Thanks, 
TMaster

### CODE REVIEW

**Comment by TMaster** [comment]
Please notice that the commit **Remove is expanding**  rollbacks the changes made in **AllowExpanding the node once** which make the change **Support user renaming with the tree-controller** the only chnage

**Comment by rychkog** [comment]
@TMaster Will look at the PR ASAP (will try to look at it today or tomorrow) :) 
@TMaster It is better to send PR to `develop` branch. Cause master supposed to be only for releases.

**Comment by TMaster** [comment]
Thanks a lot @rychkog.

I really appreciate the fast response.

I will try to finish the ngrx PR by Saturday (Need to add some tests).

and got it, I will make the PR to the `develop` branch.

Thanks a lot

File: src/tree-controller.ts [hunk 1]
```diff
@@ -85,4 +85,9 @@ export class TreeController {
       this.tree.setChildren(children);
     }
   }
+
+  public startRename(): void {
```
Comment (rychkog) [inline src/tree-controller.ts]: @TMaster should be named `startRenaming`

File: test/tree-controller.spec.ts [hunk 2]
```diff
@@ -327,14 +328,25 @@ describe('TreeController', () => {
...
+      { value: 'N1' },
+      { value: 'N2' },
     ]);
 
     fixture.detectChanges();
 
     expect(childrenOf(child).length).toEqual(0);
   });
+
+  it('Knows to transit node into BeingRenameState', () => {
```
Comment (rychkog) [inline test/tree-controller.spec.ts]: Should be `knows how to transfer a node into a BeingRenamed state`

**Comment by TMaster** [comment]
Thanks @rychkog.

I think I've addressed all your requests (regarding this PR).

Please let me know if additional refactoring is necessary.

Thanks,
TMaster

**Comment by rychkog** [comment]
@TMaster Merged :)

**Comment by rychkog** [comment]
@TMaster We will produce a new release once merge your feature with NextLevelLoading

**Comment by TMaster** [comment]
Thanks

**Pull request merged**